File: 022_Names_Scores/Solution.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var testCase int
	fmt.Scan(&testCase)

	namesList := make([]string, testCase)
	for i := 0; i < testCase; i++ {
		var name string
		fmt.Scan(&name)
		namesList[i] = name
	}

	sort.Strings(namesList)

	var q int
	fmt.Scan(&q)

	for i := 0; i < q; i++ {
		var nameQ string
		fmt.Scan(&nameQ)

		sum := 0
		for count, name := range namesList {
			if nameQ == name {
				for _, k := range name {
					sum += int(k - 'A' + 1)
				}
				sum *= count + 1
			}
		}
		fmt.Println(sum)
	}
}

File: 022_Names_Scores/Solution.py
test_case =int(input().strip())

names_list = list()

for i in range(test_case):
    names_list.append(input().strip())

names_list.sort()
q = int(input().strip())

for i in range(q):
    name_q=input().strip()
    _sum = 0
    for count,name in enumerate(names_list):
        if name_q == name:
            for k in name:
                _sum = _sum + (ord(k)-ord('A')+1)
            _sum = (count+1) * _sum
    print(_sum)